openwisp_firmware_upgrader/tasks.py

```diff
@@ -15,6 +16,60 @@
 logger = logging.getLogger(__name__)
 
 
+@shared_task(base=OpenwispCeleryTask, bind=True)
+def delete_firmware_files(self, file_paths):
+    """
+    Deletes firmware files from storage in the background
+    """
+    logger.info(f"Starting delete_firmware_files task with paths: {file_paths}")
+    storage = app_settings.PRIVATE_STORAGE_INSTANCE
+    if not storage:
+        logger.error("No storage instance configured")
+        return
+    logger.info(f"Using storage backend: {storage.__class__.__name__}")
+    logger.info(f"Storage location: {storage.location}")
+
+    for path in file_paths:
+        logger.info(f"Attempting to delete file: {path}")
+        try:
+            try:
+                exists = storage.exists(path)
+                logger.info(f"File exists: {exists}")
+                if exists:
+                    full_path = storage.path(path)
```

Comment:
What happens if you don't build the full path? What if you just delete without check if the file exists? Does it raise an exception?